src/plugin/plugin.js

```diff
@@ -553,9 +555,18 @@ class Plugin {
           this.restart = false;
           this.process.p = null;
         } else {
+          if (this.restartDelay < 30 * 1000) {
+            this.restartDelay += 1000;
+          }
+          if (this.lastRestart + 60 * 1000 < Date.now()) {
+            this.restartDelay = 0;
+          }
           console.log('Plugin:', this.pluginId, 'died, code =', code,
-                      'restarting...');
-          this.start();
+                      'restarting after', this.restartDelay);
+          this.lastRestart = Date.now();
+          setTimeout(() => {
+            this.start();
```

Comment:
`this.lastRestart` should probably be moved inside the timeout.

Reply:
:+1: